### Criando um Modelo de Classificação com o dataset Iris
Yane Sarmento Holanda
Referência: [Data Professor](https://www.youtube.com/watch?v=XmSlFPDjKdc&list=PLtqF5YXg7GLltQSLKSTnwCcHqTZASedbO)

### 1. Importando as libraries

In [2]:
from sklearn import datasets
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import make_classification

### 2. Importando o dataset

In [4]:
iris = datasets.load_iris()

### 3. Atributos

In [5]:
print(iris.feature_names)

['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [6]:
print(iris.target_names)

['setosa' 'versicolor' 'virginica']


### 4. Visão geral dos dados

In [8]:
iris.data

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

In [9]:
iris.target

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

### 4.1. Criando os atributos e os rótulos

In [10]:
X = iris.data
Y = iris.target

### 4.2. Visão geral

In [13]:
X.shape

(150, 4)

Isso significa que o X possui 150 linhas e 4 colunas

In [14]:
Y.shape

(150,)

Já o Y possui 150 linhas e 1 coluna

### 5. Criando o modelo de classificação

In [17]:
clf = RandomForestClassifier()

In [18]:
clf.fit(X,Y)

RandomForestClassifier()

### 5.1 Importância dos atributos

In [20]:
print(clf.feature_importances_)

[0.09726303 0.02776397 0.42356319 0.45140982]


### 6. Fazendo a predição

In [21]:
X[0]

array([5.1, 3.5, 1.4, 0.2])

In [22]:
print(clf.predict([[5.1, 3.5, 1.4, 0.2]]))

[0]


In [23]:
clf.fit(iris.data, iris.target_names[iris.target])

RandomForestClassifier()

### 7. Dividindo os dados

In [24]:
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2)

In [25]:
X_train.shape, Y_train.shape

((120, 4), (120,))

In [26]:
X_test.shape, Y_test.shape

((30, 4), (30,))

Dividimos as 150 linhas 4 colunas em: 120 linhas para treino e 30 linhas para teste

In [27]:
clf.fit(X_train, Y_train)

RandomForestClassifier()

In [28]:
print(clf.predict([[5.1, 3.5, 1.4, 0.2]]))

[0]


In [30]:
print(clf.predict(X_test))

[0 0 2 0 1 1 2 2 2 2 2 2 1 2 0 0 1 1 0 2 2 2 2 2 2 1 1 0 1 1]


In [31]:
print(Y_test)

[0 0 2 0 1 1 2 2 1 2 2 2 1 2 0 0 1 1 0 2 2 2 2 2 2 1 1 0 1 2]


In [29]:
print(clf.score(X_test, Y_test))

0.9333333333333333


Nosso modelo possui 93% de chances de acerto